### PR TITLE
Team 1  User can add items in cart and checkout test And User can sort products test

### PR DESCRIPTION
**Pull request by Kristinochka-bee**

### CODE REVIEW

File: pages/cart/CartPage.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import {PageObject} from '../PageObject';
+import {Locator, Page} from '@playwright/test';
+import { expect} from '@playwright/test';
+
+export class CartPage extends PageObject {
+  protected readonly continueShoppingButton: Locator;
+  protected readonly checkoutButton: Locator;
+  protected readonly removeButton: Locator;
+  protected readonly yourCartTitle: Locator;
```
Comment (ABalitskaya) [inline pages/cart/CartPage.ts]: This locator is not described in the constructor

File: pages/cart/CartPage.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  }
+
+  async openProductPage(cardName: string) {
+    await this.page.locator(`${cardName}`).locator("//div[@class='inventory_item_name']").click();
+  }
+  async clickOnCheckoutButton() {
+    await this.checkoutButton.click();
+  }
+
+  async checkUserOnCartPage() {
```
Comment (ABalitskaya) [inline pages/cart/CartPage.ts]: This method has an undescribed locator. Will you use this method in the test?

File: pages/checkout/CheckoutPage.ts [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import {PageObject} from '../PageObject';
+import {Locator, Page} from '@playwright/test';
+
+export class CheckoutPage extends PageObject {
+  protected readonly firstNameField: Locator;
+  protected readonly lastNameField: Locator;
+  protected readonly zipPostalCodeField: Locator;
+  protected readonly continueButton: Locator;
+  protected readonly checkoutYourInformation: Locator;
```
Comment (ABalitskaya) [inline pages/checkout/CheckoutPage.ts]: unused locator

File: e2e/products/cart/addProductInCartAndCheckout.spec.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    const cartPage = new CartPage(page);
+    const overviewPage =  new OverviewPage(page);
+    const checkoutPage = new CheckoutPage(page);
+    const completePage = new CompletePage(page);
+
+    await productsPage.addToProduct('Sauce Labs Bike Light');
+    await productsPage.addToProduct('Sauce Labs Bolt T-Shirt');
+    await productsPage.addToProduct('Sauce Labs Onesie');
+
+    await productsPage.clickOnShoppingCartLink();
```
Comment (ABalitskaya) [inline e2e/products/cart/addProductInCartAndCheckout.spec.ts:32]: You need to check that list of products is correct

File: e2e/products/cart/addProductInCartAndCheckout.spec.ts [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    await cartPage.removeProduct('Sauce Labs Onesie');
```
Comment (ABalitskaya) [inline e2e/products/cart/addProductInCartAndCheckout.spec.ts]: After removing you need to check that product is removed

File: pages/cart/CartPage.ts [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+import { expect} from '@playwright/test';
+
+export class CartPage extends PageObject {
+  protected readonly continueShoppingButton: Locator;
+  protected readonly checkoutButton: Locator;
+  protected readonly removeButton: Locator;
+  protected readonly yourCartTitle: Locator;
+
+  constructor(page: Page) {
+    super(page, '/');
```
Comment (ABalitskaya) [inline pages/cart/CartPage.ts:14]: You need to write the correct actual URL

File: pages/overview/OverviewPage.ts [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+export class OverviewPage extends PageObject {
+  protected readonly overviewTitle: Locator;
+  protected readonly itemTotal: Locator;
+  protected readonly taxItem: Locator;
+  protected readonly total: Locator;
+  protected readonly finishButton: Locator;
+  protected readonly cancelButton: Locator;
+  protected readonly itemsPrice: Locator;
+  constructor(page: Page) {
+    super(page, '/');
```
Comment (ABalitskaya) [inline pages/overview/OverviewPage.ts:18]: You need to write the correct actual URL

File: e2e/products/cart/addProductInCartAndCheckout.spec.ts [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+    await cartPage.clickOnCheckoutButton();
+    await expect(page).toHaveURL('/checkout-step-one.html');
+    await checkoutPage.fillUsersField('Alex', 'Spigel', '44444');
```
Comment (ABalitskaya) [inline e2e/products/cart/addProductInCartAndCheckout.spec.ts]: Better to write data to variables

File: e2e/products/cart/addProductInCartAndCheckout.spec.ts [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+  test('User can add items in cart and checkout', async ({page}) => {
+    await loginTestHelper(page, 'standard_user');
+
+    const productsPage = new ProductsPage(page);
+    const cartPage = new CartPage(page);
+    const overviewPage =  new OverviewPage(page);
+    const checkoutPage = new CheckoutPage(page);
+    const completePage = new CompletePage(page);
+
+    await productsPage.addToProduct('Sauce Labs Bike Light');
```
Comment (ABalitskaya) [inline e2e/products/cart/addProductInCartAndCheckout.spec.ts]: Better to use it: 
await productsPage.addToProduct(productNames[i]);
where [i] is index of current product

File: e2e/products/cart/addProductInCartAndCheckout.spec.ts [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+    await productsPage.addToProduct('Sauce Labs Bolt T-Shirt');
```
Comment (ABalitskaya) [inline e2e/products/cart/addProductInCartAndCheckout.spec.ts]: Better to use it: 
await productsPage.addToProduct(productNames[i]);
where [i] is index of current product

File: e2e/products/cart/addProductInCartAndCheckout.spec.ts [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+    await productsPage.addToProduct('Sauce Labs Onesie');
```
Comment (ABalitskaya) [inline e2e/products/cart/addProductInCartAndCheckout.spec.ts]: Better to use it: 
await productsPage.addToProduct(productNames[i]);
where [i] is index of current product

File: e2e/products/cart/addProductInCartAndCheckout.spec.ts (hunk 5 above)
Comment (ABalitskaya) [inline e2e/products/cart/addProductInCartAndCheckout.spec.ts]: Better to use it: 
await cartPage.removeProduct(productNames[i]);
where [i] is index of current product

File: e2e/products/cart/addProductInCartAndCheckout.spec.ts [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+import {CheckoutPage} from "../../../pages/checkout/CheckoutPage";
+import { expect} from '@playwright/test';
+import {CompletePage} from "../../../pages/сomplete/CompletePage";
+
+test.describe('Add product in cart and checkout', async () => {
+  test('User can add items in cart and checkout', async ({page}) => {
+    await loginTestHelper(page, 'standard_user');
+
+    const productsPage = new ProductsPage(page);
+    const cartPage = new CartPage(page);
```
Comment (ABalitskaya) [inline e2e/products/cart/addProductInCartAndCheckout.spec.ts:20]: Better to declare new pages in the part of your code where they are used for the first time

File: e2e/products/cart/addProductInCartAndCheckout.spec.ts (hunk 9 above)
Comment (ABalitskaya) [inline e2e/products/cart/addProductInCartAndCheckout.spec.ts]: When you want to add more than one product better to use cycle.

File: pages/cart/CartPage.ts (hunk 1 above)
Comment (Kristinochka-bee) [reply pages/cart/CartPage.ts]: fixed

File: pages/cart/CartPage.ts (hunk 2 above)
Comment (Kristinochka-bee) [reply pages/cart/CartPage.ts]: done

File: pages/checkout/CheckoutPage.ts (hunk 3 above)
Comment (Kristinochka-bee) [reply pages/checkout/CheckoutPage.ts]: done

File: e2e/products/cart/addProductInCartAndCheckout.spec.ts (hunk 10 above)
Comment (Kristinochka-bee) [reply e2e/products/cart/addProductInCartAndCheckout.spec.ts]: done

**Pull request merged**